### PR TITLE
Added the delegate operator article

### PR DESCRIPTION
**Pull request by pkulikov**
Contributes to #10741 
Supported by dotnet/samples#1093

### CODE REVIEW

File: docs/csharp/language-reference/operators/index.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ These are the highest precedence operators.
 
 [nameof](nameof.md) - obtains the simple (unqualified) name of a variable, type, or member as a constant string.
 
-[delegate](../../programming-guide/statements-expressions-operators/anonymous-methods.md) – declares and returns a delegate instance.
```
Comment (pkulikov) [inline docs/csharp/language-reference/operators/index.md]: @BillWagner though this programming guide article provides more trivia (which can found in the spec) about the anonymous methods, I suggest to remove it and redirect to the newly created article about the `delegate` operator. So, that one to remove:
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/anonymous-methods

If that's fine with you, I'll do that in the separate PR.

Comment (BillWagner) [reply docs/csharp/language-reference/operators/index.md]: I like that idea. We should minimize coverage of using the `delegate` syntax. It's not preferred anymore.

File: docs/csharp/language-reference/operators/delegate-operator.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  - "delegate [C#]"
+  - "anonymous method [C#]"
+---
+# delegate operator (C# reference)
+
+The `delegate` operator creates an anonymous method that can be converted to a delegate type:
+
+[!code-csharp-interactive[anonymous method](~/samples/csharp/language-reference/operators/DelegateOperator.cs#AnonymousMethod)]
+
+Beginning with C# 3, [lambda expressions](../../programming-guide/statements-expressions-operators/lambda-expressions.md) provide a more concise and expressive way to create an anonymous function. Use the [=> operator](lambda-operator.md) to construct a lambda expression:
```
Comment (BillWagner) [inline docs/csharp/language-reference/operators/delegate-operator.md]: I like that idea.

This article should more clearly reference the discussion on capturing outer variables and closures in the article on [lambda expressions](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/lambda-expressions) in the programming guide. 

Consider adding a sentence specifically mentioning that discussion.

**Review by BillWagner — COMMENTED** [review]
This is good. I like how we're building a guide that encourages more modern practices.

I like your suggestion, and had one of my own. Then, I think this is ready.

File: docs/csharp/language-reference/operators/delegate-operator.md (hunk 2 above)
Comment (pkulikov) [reply docs/csharp/language-reference/operators/delegate-operator.md]: @BillWagner please take a look at my attempt made in the latest commit. If it's fine with you, let's merge it, I'll update the programming guide in the next PR.

**Review by BillWagner — APPROVED** [review]
I like this. It's a great step toward more modern guidance.

I'll :shipit: now.

**Pull request merged**